lib/iris/tests/test_constraints.py

```diff
@@ -295,50 +296,241 @@ def load_match(self, files, constraints):
         return cubes
 
 
+@tests.skip_data
+class TestCubeExtract__names(TestMixin, tests.IrisTest):
+    def setUp(self):
+        TestMixin.setUp(self)
+        self.cube = iris.load_cube(self.theta_path)
+        # Expected names...
+        self.standard_name = "air_potential_temperature"
+        self.long_name = "air potential temperature"
```

Comment:
@pp-mo I think that you misunderstand. The `self.theta_path` loads a `cube` that already has the `standard_name` and `STASH` set, but the `long_name` and `var_name` are both `None`.

The `setUp` is hardwiring in those missing `long_name` and `STASH` values to the loaded cube, so that the tests all have meaningful values to constrain against.

Reply:
I didn't mean that it is pointless to set long_name, but that it is pointless to set it to *the same as standard_name*.
Since `test_long_name` uses `Constraint(self.long_name)`, this passes "by coincidence" as that string also matches `cube.standard_name` : it is not testing comparison with `cube.long_name`.